bamt/builders/evo_builder.py

```diff
@@ -0,0 +1,197 @@
+
+        # Define the requirements for the evolutionary algorithm
+        requirements = GraphRequirements(
+            max_arity=kwargs.get(
+                'max_arity', self.default_max_arity), max_depth=kwargs.get(
+                'timeout', self.default_max_depth),
+            timeout=timedelta(
+                seconds=kwargs.get(
+                    'timeout', self.default_timeout)),
+            n_jobs=kwargs.get('n_jobs', self.default_n_jobs))
+
+        # Set the parameters for the evolutionary algorithm
+        optimizer_parameters = GPAlgorithmParameters(
+            pop_size=kwargs.get('pop_size', self.default_pop_size),
+            crossover_prob=kwargs.get('crossover_prob', self.default_crossover_prob),
+            mutation_prob=kwargs.get('mutation_prob', self.default_mutation_prob),
+            genetic_scheme_type=GeneticSchemeTypesEnum.steady_state,
+            mutation_types=kwargs.get('custom_mutations', self.default_mutations),
+            crossover_types=kwargs.get('custom_crossovers', self.default_crossovers),
+            selection_types=kwargs.get('selection_type', self.default_selection))
+
+        # Set the adapter for the conversion between the graph and the data
+        # structures used by the optimizer
+        adapter = DirectAdapter(
+            base_graph_class=evo.CustomGraphModel,
+            base_node_class=evo.CustomGraphNode)
+
+        # Set the constraints for the graph
+
+        if kwargs.get('blacklist', None) is not None:
+            self.default_constraints.append(evo.has_no_blacklist_edges)
+        if kwargs.get('whitelist', None) is not None:
+            self.default_constraints.append(evo.has_only_whitelist_edges)
+
+        constraints = kwargs.get(
+            'custom_constraints',
+            self.default_constraints)
+
+        graph_generation_params = GraphGenerationParams(
+            adapter=adapter,
+            rules_for_constraint=constraints,
+            available_node_types=nodes_types)
+
+        # Define the objective function to optimize
+        objective = Objective({'custom': kwargs.get(
+            'custom_metric', self.objective_metric)})
+
+        # Initialize the optimizer
+        optimizer = EvoGraphOptimizer(
+            objective=objective,
+            initial_graphs=initial,
+            requirements=requirements,
+            graph_generation_params=graph_generation_params,
+            graph_optimizer_params=optimizer_parameters)
+
+        # Define the function to evaluate the objective function
+        objective_eval = ObjectiveEvaluate(
+            objective, data=data)
+
+        # Run the optimization
+        optimized_graph = optimizer.optimise(objective_eval)[0]
+
+        # Get the best graph
+        best_graph_edge_list = optimized_graph.operator.get_edges()
+        best_graph_edge_list = self._convert_to_strings(best_graph_edge_list)
+
+        print('Best graph: ', best_graph_edge_list)
```

Comment:
I think, the better way here is to save it somewhere, or at least add a flag to control verbosity

Reply:
It was temporary of debugging, I'll remove it